assets/src/domain/eventEditor/ui/tickets/ticketPriceCalculator/fields/MoneyField/updateInputsSizes.tsx

```diff
@@ -0,0 +1,39 @@
+interface updatedPercentSignInputProps {
+	beforeWidth: number;
+	inputWidth: number;
+	signB4: boolean;
+}
+
+const updatedPercentSignInput = ({ beforeWidth, inputWidth, signB4 }: updatedPercentSignInputProps): void => {
+	document
+		.querySelectorAll('.ee-money-field--with-percent-sign .ee-input__price-field')
+		.forEach((element: HTMLElement) => {
+			if (signB4) {
+				element.style.marginLeft = `${beforeWidth}px`;
+			}
+
+			element.style.width = `${inputWidth}px`;
+		});
```

Comment:
whoah... why are you adding styles like this?
why not within the component when it renders?

Reply:
the data flow is a bit tricky because first it should check sizes from a different row and then pass back down to input levels the sizes

Reply:
that shouldn't be necessary because the offset is based on the currency sign and not the inputs
technically we should be able to determine the offset before the TPC even loads.

there are basically three scenarios to account for, but the first won't ever happen, so only the last two scenarios from the following cases will actually exist:

- currency sign before amount : prepend inputs with space for currency sign

- currency sign before + percentage modifiers : same as above, but need to also append inputs with space for % sign. The previous point won't actually exist because we will almost always need to account for %, so may as well just do so by default

- currency sign after amount : append inputs with space for currency sign. the same spacer will be used for % signs as well, so if the currency sign is 3 characters, then the spacer for the % sign will also be the same width as if it had 3 characters.

So to describe those two real scenarios again but in a different way:

- currency sign before amount: add spacer before all inputs with width defined by number of characters in currency sign PLUS add single width spacer after all inputs for % sign:  `[$$$$][ input ][%]`

- - currency sign after amount: add spacer after all inputs with width defined by number of characters in currency sign, same spacer will be used for % sign:  `[ input ][$%$%]`

Reply:
yes, after I've done that direct DOM mumbo-jumbo also was thinking that it's simpler to derive the space needed from sign length; was focused on doing that refactor related to BaseField and chakra's inputs, but will focus now on the aspect you've described